maplibre/src/render/eventually.rs

```diff
@@ -47,9 +47,16 @@ where
 }
 impl<T> Eventually<T> {
     #[tracing::instrument(name = "initialize", skip_all)]
-    pub fn initialize(&mut self, f: impl FnOnce() -> T) {
+    pub fn initialize(&mut self, f: impl FnOnce() -> T) -> &T {
         if let Eventually::Uninitialized = self {
-            *self = Eventually::Initialized(f());
+            if let Eventually::Uninitialized = self {
```

Comment:
Seems like this conditional check is unnecessarily duplicated.

Reply:
As small aside, this seems a bit like LazyCell which has recently been stabilized, if you wanted to use something from std. I might be wrong though, I'm not familiar with your codebase.

Reply:
ah lol yeah merge issue :) thanks!

I think LazyCell is not ideal because it does not really have that uninitialized state without giving it already a lambda what the value is going to be.

Basically Eventually is a slightly more fancy Option